maze.py

```diff
@@ -0,0 +1,162 @@
+import sys
+import pygame
+import threading
+from pharmacontroller import SCREEN_SIZE, PharmaScreen
+from textwriter import String
+
+# Constants for the game
+TRAIL_TIME = 5  # Time in seconds before trail disappears
+
+game_started = False
+game_over = False
+
+# Directions (example level input)
+level_directions = [
+   "right", "right", "right", "right", "right", "right", "up", "right", "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "left", "up", "up", "right", "right",
+   "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "up", "right", "right", "right", "up", "up", "left", "up", "left", "left", "left", "left", "left", "left", "left",
+   "up", "up", "up", "up", "up", "up", "up", "up", "left", "left", "left", "up", "up", "up", "up", "up", "up", "up", "up"
+]
+# To create level, copy the array above with the same name and replace the directions with the correct ones
+
+# Colors
+TRAIL_COLOR = 1.0  # Using 1.0 value for 'on' pixels
+EMPTY_COLOR = 0.0  # Using 0.0 value for 'off' pixels
+
+pygame.init()
+
+# Define movement directions as x, y deltas
+DIRECTIONS = {
+    "up": (0, -1),
+    "down": (0, 1),
+    "left": (-1, 0),
+    "right": (1, 0),
+}
+
+class Player:
+    def __init__(self, start_x, start_y):
```

Comment:
Using a [dataclass](https://docs.python.org/3/library/dataclasses.html) with a [__post_init__](https://docs.python.org/3/library/dataclasses.html#dataclasses.__post_init__) function to initialize values that depend on others would be better I think
It would look like this

```Python
from dataclasses import dataclass, field
from typing import List, Tuple

@dataclass
class Player:
    start_x: int
    start_y: int
    trail_directions: List[Tuple[int, int]] = field(default_factory=list)
    trail: List[Tuple[int, int]] = field(default_factory=list)
    x: int = field(init=False)
    y: int = field(init=False)

    def __post_init__(self):
        self.x = self.start_x
        self.y = self.start_y
       self.trail.append((self.x, self.y))
```